src/zkevm_specs/evm/execution/memory.py

```diff
@@ -0,0 +1,41 @@
+from ..instruction import Instruction, Transition
+from ..opcode import Opcode
+from ..table import RW, CallContextFieldTag
+from ...util import FQ
+
+
+def memory(instruction: Instruction):
+    opcode = instruction.opcode_lookup(True)
+
+    address = instruction.stack_pop()
+
+    is_mload = instruction.is_equal(opcode, Opcode.MLOAD)
+    is_mstore8 = instruction.is_equal(opcode, Opcode.MSTORE8)
+    is_store = FQ(1) - is_mload
+    is_not_mstore8 = FQ(1) - is_mstore8
+
+    value = instruction.stack_push() if is_mload == FQ(1) else instruction.stack_pop()
+
+    src_id = instruction.call_context_lookup(CallContextFieldTag.TxId)
```

Comment:
I didn't find this lookup in the circuit.  I see that you use it in the `memory_lookup`, but that function takes a `call_id` and not a `tx_id`.  So I think this should be removed, and the `memory_lookup` for `is_not_mstore8` should get the 3rd parameter removed (the `src_id`).